torch/_inductor/runtime/benchmarking.py

```diff
@@ -7,9 +8,11 @@
 
 import torch
 from torch._dynamo.utils import counters, dynamo_timed
+from torch._inductor.config import use_experimental_benchmarker
 
 
 logger = torch._logging.getArtifactLogger(__name__, "benchmarking")
+use_experimental_benchmarker = use_experimental_benchmarker and not torch.xpu._is_compiled
```

Comment:
@etaf we're going to force disable the experimental benchmarker on xpu for now; it's not so straightforward to figure out if we're on xpu when `benchmark_gpu` is called since the tensors are already wrapped away

Reply:
@nmacchioni  thanks for your information, but I suggest using the common device interface in `torch/_dynamo/device_interface.py` so that the code is more generic and will easy to enable for other GPU.

Reply:
@etaf we can't use `torch/_dynamo/device_interface.py` since that requires `device_type`, we don't know `device_type` in `benchmark_gpu` since `_callable` is essentially a black box kernel. We'd either have to migrate all call sites to `benchmark` or modify them to pass `device_type` to `benchmark_gpu`, neither of which I am interested in doing at the moment.

Reply:
@nmacchioni  Maybe use `get_gpu_type()` in `torch/_inductor/utils.py` ?

Reply:
@etaf Under the hood is this not more or less `torch.xpu._is_compiled()`?

Regardless, I don't feel that it is particularly safe to enable this outside of Nvidia/AMD hardware at the moment; there is no guarantee that this specialized benchmarking module will work as expected on other hardware, and I certainly haven't done the required performance for any other hardware.

Reply:
@nmacchioni , agree with you that it only works on CUDA at the moment, what I meant was that using a common device interface would make the code more generic and extensible, not that it should work on other GPUs now. If this is going to increase your work, we can wait to do the generalization work later when other GPUs need it ^_^.

Reply:
@etaf Sounds good! We can always work on an XPU focused benchmarked in the future, should only require some slight algorithmic tweaks